### DISCUSSION TITLE
Longer time-to-readiness with Chromium 110

### DISCUSSION THREAD

**Discussion by gorhill** [uBlockOrigin/uAssets#16939] [General]
I notice a steep regression in time-to-readiness starting with Chromium 110.

I can see this on my side and this appears to be confirmed by looking at _Support_ => `allReadyAfter` for numerous filter issues with Chromium 110 being mentioned. This has to be a change in Chromium itself, since any previous version of Chromium do not systematically report 3000+ ms for `allReadyAfter` despite the same uBO version being used.

This means there could be an uptick in reports of Youtube video ads not being blocked or similar cases at browser launch.

**Comment by stephenhawk8054** [comment]
Do we have any performance measuring tools to see what processes are taking time for the time-to-readiness like profiler?

**Reply by gorhill** [reply]
I can't pinpoint exactly where in Chromium the exact regression is performance-wise, but from what I have observed it does look like it's `browser.storage`-related. The regression in time-to-readiness occurs only at browser launch, i.e. it does not occur if you manually start uBO after the browser is already launched, in which case I get a time-to-readiness of 125 ms (selfie), while I am currently getting over 2000 ms (selfie) at browser launch -- despite the exact same sequence of code being executed.

**Reply by stephenhawk8054** [reply]
Do you see the same slowness with other extensions (especially other blockers...)? I suspect if this affects extensions in general, users who use more extensions will feel the slow starting more obviously (currently my chromium browser only has uBO and I haven't seen the slowness to the point of over 10000ms like in some issues reported).

---

Or it might relate to history/storage as my chromium is just used for testing, not for real using.

**Reply by gorhill** [reply]
> Do you see the same slowness with other extensions (especially other blockers...)?

I see the same behavior in AdGuard and ABP. My test is to load `https://arstechnica.com/` in a tab, quit the browser, then launch the browser with that tab active. With any blockers, you can see that the block count keeps climbing, because the ad/tracker scripts were not actually blocked and these causes a lot of requests to be blocked once the blocker is fully initialized, even though cosmetically all seem to be blocked. If you force a reload of the page when you think the extension is fully initialized, then the block count should stay stable in the low 2-digit.

**Comment by garry-ut99** [comment]
1) https://bugs.chromium.org/p/chromium/issues/detail?id=374592
2) https://groups.google.com/a/chromium.org/g/chromium-extensions/c/_ks4MjNLlN8

**Comment by uBlock-user** [comment]
https://www.reddit.com/r/uBlockOrigin/comments/11fj3fa/ublock_takes_long_time_to_load/

allReadyAfter: 249271 ms

**Reply by garry-ut99** [reply]
More big and even bigger cases (Chromium 110 and above) : 

- https://www.reddit.com/r/uBlockOrigin/comments/113ck2e/comment/j8pen18/  `unknown (not ready)`
- https://github.com/uBlockOrigin/uAssets/issues/16773 `unknown (not ready)`
- https://github.com/uBlockOrigin/uAssets/issues/16890 `unknown (not ready)`
- https://github.com/uBlockOrigin/uAssets/issues/16932 `unknown (not ready)`
- https://github.com/uBlockOrigin/uAssets/issues/16992 `unknown (not ready)`
- https://github.com/uBlockOrigin/uAssets/issues/17035 `unknown (not ready)`
- https://github.com/uBlockOrigin/uAssets/issues/16813 `unknown (not ready)`
- https://github.com/uBlockOrigin/uAssets/issues/13709 `unknown (not ready)` (102)
- https://github.com/uBlockOrigin/uAssets/issues/20795 `allReadyAfter: 371461 ms (selfie)`, `maxAssetCacheWait: 8312 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW BIG 3!**
- https://github.com/uBlockOrigin/uAssets/issues/19888 `allReadyAfter: 285623 ms (selfie)`, `maxAssetCacheWait: 1774 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW BIG 3!**
also https://github.com/uBlockOrigin/uAssets/issues/20080
- https://github.com/uBlockOrigin/uAssets/issues/18464 `allReadyAfter: 282776 ms (selfie)`, `maxAssetCacheWait: 16459 ms`
found first by `dportvine`,  see : https://github.com/uBlockOrigin/uAssets/discussions/16939#discussioncomment-6145449
- https://www.reddit.com/r/uBlockOrigin/comments/11fj3fa/ublock_takes_long_time_to_load/ `allReadyAfter: 249271 ms (selfie)`, `maxAssetCacheWait: 156 ms`
found first by `uBlock-user,` see : https://github.com/uBlockOrigin/uAssets/discussions/16939#discussioncomment-5176293
- https://github.com/uBlockOrigin/uAssets/issues/16752 `allReadyAfter: 248926 ms (selfie)`, `maxAssetCacheWait: 290 ms`
- https://github.com/uBlockOrigin/uAssets/issues/17290 `allReadyAfter: 195139 ms (selfie)`, `maxAssetCacheWait: 348 ms` (111) █ **NEW BIG!**
- https://github.com/uBlockOrigin/uAssets/issues/16927 `allReadyAfter: 133151 ms (selfie)`, `maxAssetCacheWait: 161 ms`
- https://github.com/uBlockOrigin/uAssets/issues/18623 `allReadyAfter: 102064 ms (selfie)`, `maxAssetCacheWait: 517 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW BIG 3!**
- https://github.com/uBlockOrigin/uAssets/issues/23194 `allReadyAfter: 90957 ms`, `maxAssetCacheWait: 4432 ms` (121) ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW BIG 4!**
- https://github.com/uBlockOrigin/uAssets/issues/16738 `allReadyAfter: 89562 ms (selfie)`, `maxAssetCacheWait: 2019 ms`
- https://github.com/uBlockOrigin/uAssets/issues/23344 `allReadyAfter: 85031 ms	(selfie)`, `maxAssetCache: 6043 ms` (121) ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW BIG 4!**
- https://github.com/uBlockOrigin/uAssets/issues/22809 `allReadyAfter: 79347 ms`, `maxAssetCacheWait: 23669 ms` (114) ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW BIG 4!**
- https://github.com/uBlockOrigin/uAssets/issues/23312 `allReadyAfter: 72803 ms	(selfie)`, `maxAsset: 9986 ms` (121) ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW BIG 4!**
- https://github.com/uBlockOrigin/uAssets/issues/23261 `allReadyAfter: 69402 ms	(selfie)`, `maxAssetCacheWait: 1768 ms` (123) ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW BIG 4!**
- https://github.com/uBlockOrigin/uAssets/issues/17215 `allReadyAfter: 63866 ms (selfie)`, `maxAssetCacheWait: 534 ms` (111)
- https://github.com/uBlockOrigin/uAssets/issues/18554 `allReadyAfter: 58783 ms (selfie)`, `maxAssetCacheWait: 7684 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW BIG 3!**
- https://github.com/uBlockOrigin/uAssets/issues/17129 `allReadyAfter: 55924 ms (selfie)`,  `maxAssetCacheWait: 143 ms` (111)
- https://github.com/uBlockOrigin/uAssets/issues/17542 `allReadyAfter: 54162 ms (selfie)`,  `maxAsset: 554 ms` (111) █ **NEW**
- https://github.com/uBlockOrigin/uAssets/issues/17128 `allReadyAfter: 50254 ms (selfie)`,  `maxAssetCacheWait: 579 ms` (111)
- https://github.com/uBlockOrigin/uAssets/issues/17136 `allReadyAfter: 50254 ms (selfie)`,  `maxAssetCacheWait: 579 ms` (111)
- https://github.com/uBlockOrigin/uAssets/issues/21063 `allReadyAfter: 46817 ms	(selfie)`, `maxAssetCache: 460 ms` (119) ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW 4!**
- https://github.com/uBlockOrigin/uAssets/issues/16958 `allReadyAfter: 43253 ms (selfie)`,  `maxAssetCacheWait: 99 ms`
- https://github.com/uBlockOrigin/uAssets/issues/22488 `allReadyAfter: 41197 ms	(selfie)`, `maxAssetCacheWait: 205 ms` (121) ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW 4!**
- https://github.com/uBlockOrigin/uAssets/issues/22717 `allReadyAfter: 40098 ms	(selfie)`, `maxAssetCacheWait: 1872 ms` (123) ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW 4!**
- https://github.com/uBlockOrigin/uAssets/issues/20137 `allReadyAfter: 37565 ms (selfie)`, `maxAssetCacheWait: 651 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/18180 `allReadyAfter: 36481 ms (selfie)`, `maxAs: 6267 ms` (112) ![#1589F0](https://placehold.co/15x15/1589F0/1589F0.png) **NEW 2**
- https://github.com/uBlockOrigin/uAssets/issues/19229 `allReadyAfter: 34592 ms (selfie)`, `maxAssetCacheWait: 835 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/23066 `allReadyAfter: 34034 ms	(selfie)`, `maxAssetCacheWait: 2173 ms` (123) ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW 4!**
- https://github.com/uBlockOrigin/uAssets/issues/18261 `allReadyAfter: 31723 ms (selfie)`, `maxAssetCacheWait: 80 ms` (112) ![#1589F0](https://placehold.co/15x15/1589F0/1589F0.png) **NEW 2**
- https://github.com/uBlockOrigin/uAssets/issues/21608 `allReadyAfter: 31077 ms	(selfie)`, `maxAssetCacheWait: 1192 ms` (120)  ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW 4!**
- https://github.com/uBlockOrigin/uAssets/issues/17257 `allReadyAfter: 31020 ms (selfie)`, `maxAssetCacheWait: 118 ms` (111)
- https://github.com/uBlockOrigin/uAssets/issues/17102 `allReadyAfter: 27897 ms (selfie)`, `maxAssetCacheWait: 390 ms`
- https://github.com/uBlockOrigin/uAssets/issues/20844 `allReadyAfter: 27704 ms (selfie)`, `maxAssetCacheWait: 1019 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/19568 `allReadyAfter: 27522 ms (selfie)`, `maxAssetCacheWait: 522 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/18742 `allReadyAfter: 27061 ms (selfie)`, `maxAssetCacheWait: 299 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/20611 `allReadyAfter: 26932 ms (selfie)`, `maxAssetCacheWait: 386 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/20298 `allReadyAfter: 25780 ms (selfie)`, `maxAssetCacheWait: 678 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
also https://github.com/uBlockOrigin/uAssets/issues/20284 + https://github.com/uBlockOrigin/uAssets/issues/20283
- https://github.com/uBlockOrigin/uAssets/issues/20553 `allReadyAfter: 25592 ms (selfie)`, `maxAssetCacheWait: 2655 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/17239 `allReadyAfter: 24410 ms (selfie)`, `maxAssetCache: 2169 ms`
- https://github.com/uBlockOrigin/uAssets/issues/21028 `allReadyAfter: 24321 ms (selfie)`, `maxAssetCacheWait: 125 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/19124 `allReadyAfter: 23906 ms (selfie)`, `maxAssetCacheWait: 146 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/19865 `allReadyAfter: 22977 ms (selfie)`, `maxAssetCacheWait: 2964 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/18670 `allReadyAfter: 21933 ms (selfie)`, `maxAssetCacheWait: 760 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/16979 `allReadyAfter: 21702 ms`,  `maxAssetCacheWait: 8085 ms`
- https://github.com/uBlockOrigin/uAssets/issues/19610 `allReadyAfter: 20882 ms (selfie)`, `maxAssetCacheWait: 6838 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/17267 `allReadyAfter: 20759 ms (selfie)`,  `max: 1909 ms` (111) █ **NEW**
- https://github.com/uBlockOrigin/uAssets/issues/17262 `allReadyAfter: 20329 ms (selfie)`,  `max: 361 ms` (111) █ **NEW** + [X](17263)
- https://github.com/uBlockOrigin/uAssets/issues/19394 `allReadyAfter: 20160 ms (selfie)`, `maxAssetCacheWait: 795 ms` ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3**
- https://github.com/uBlockOrigin/uAssets/issues/17484 `allReadyAfter: 18763 ms (selfie)`,  `maxAssetCacheWait: 1445 ms` (111) █ **NEW**
- https://github.com/uBlockOrigin/uAssets/issues/17104 `allReadyAfter: 17711 ms (selfie)`,  `maxAssetCacheWait: 134 ms` (111)
- https://github.com/uBlockOrigin/uAssets/issues/16974 `allReadyAfter: 15480 ms (selfie)`,  `maxAssetCacheWait: 400 ms`
- https://github.com/uBlockOrigin/uAssets/issues/17268 `allReadyAfter: 14777 ms (selfie)`,  `maxAssetCac: 647 ms` (111) █ **NEW**
- https://github.com/uBlockOrigin/uAssets/issues/17068 `allReadyAfter: 13528 ms (selfie)`,  `maxAssetCacheWait: 1667 ms`
- https://github.com/uBlockOrigin/uAssets/issues/17281 `allReadyAfter: 13049 ms (selfie)`,  `maxAsset: 1175 ms` (111) █ **NEW**
- https://github.com/uBlockOrigin/uAssets/issues/17364 `allReadyAfter: 12915 ms`,  `maxAssetCac: 387 ms` (111) █ **NEW**
- https://github.com/uBlockOrigin/uAssets/issues/17025 `allReadyAfter: 12660 ms`,  `maxAssetCacheWait: 4142 ms` _____________█ **NEW**
- https://github.com/uBlockOrigin/uAssets/issues/17091 `allReadyAfter: 12455 ms`,  `maxAssetCac: 8650 ms` (111) █ **NEW**
- https://github.com/uBlockOrigin/uAssets/issues/17400 `allReadyAfter: 13889 ms`,  `maxAssetCacheWait: 5257 ms` (111) (Mobile) __________█ **NEW**

What a massive `maxAssetCacheWait` (Chromium 110):
- #17113 
   - `allReadyAfter: 36041 ms, (selfie)`,  `maxAssetCacheWait: 12510855 ms` - almost 3,5 hour
- #12666
   - `allReadyAfter: 327 ms (selfie)`, `maxAssetCacheWait: 6703011 ms` (99) - almost 2 hours


As for the cases: `unknown (not ready)` , they are generally related to Chromium 110, as there is only a single case reported from the past (Chromium 102), most likely if users waited enought, it would eventually be converted into some pretty big number.

As for the big numbers cases, indeed, ammount of such cases has hugely incerased since Chromium 110, however similar cases were already occurring in the past for both Chromium and Firefox, but not as often as now.

Also, we don't know about unreported cases which maybe many more of them.

The other cases you can find in the previous thread: 
- https://github.com/uBlockOrigin/uBlock-issues/issues/1865

<hr>

**New update 1 (14-04-2023)** - see entries labeled black color: " █ **NEW** "

<hr>

**New update 2 (12-06-2023)** - see entries labeled blue color : " ![#1589F0](https://placehold.co/15x15/1589F0/1589F0.png) **NEW 2** "

<hr>

**New update 3 (05-12-2023)** - see entries labeled orange coloer : " ![#FFA500](https://placehold.co/15x15/FFA500/FFA500.png) **NEW 3** " 
- only cases above 20000 ms, as there were too many cases between 10000-20000 range
- as seen there are several new big cases, 
   - the new record is `371461 ms` (increase by 1/3 over the previous record)
   - the new 2nd place is a case which is slightly ahead of its predecessor (the previous record)  `282776 ms ==> 285623 ms`
   - the two other big cases are: `102064 ms` (it's predecessor was `89562 ms` ) and `58783 ms`
- the other thread has been updated too (including another several big cases and records) :
https://github.com/uBlockOrigin/uBlock-issues/issues/1865#issuecomment-1427103640

<hr>
	
**New update 4 (26-04-2024):** - see entries labeled green color : " ![#32CD32](https://placehold.co/15x15/32CD32/32CD32.png) **NEW 4** ",  including several big cases: 
- 9 cases, 5 of them withing range `69000 ms - 91000 ms`
- the other thread has been updated too (including another several big cases and records) :
https://github.com/uBlockOrigin/uBlock-issues/issues/1865#issuecomment-1427103640

**Reply by ghost** [reply]
https://www.reddit.com/r/uBlockOrigin/comments/11gkra9/ublock_takes_time_to_work/

**Reply by uBlock-user** [reply]
He didnt mention Chromium version. The issue here is specific to 110 only.

**Reply by dportvine** [reply]
`allReadyAfter: 282776 ms (selfie)`, `maxAssetCacheWait: 16459 ms` https://github.com/uBlockOrigin/uAssets/issues/18464

**Reply by gorhill** [reply]
That person was subscribed to tons of lists, including some which are no longer in existence (try `https://gitcdn.xyz/repo/NanoAdblocker/NanoFilters/master/NanoFilters/NanoBase.txt` in browser), unresponsive server is what caused this issue.

**Comment by u-RraaLL** [comment]
Should there be an announcement made about this on reddit maybe? Should we advise people to downgrade to 109?

Has anybody tested if the issue persists into 111-113?

**Reply by gorhill** [reply]
> Should we advise people to downgrade to 109

Never should we advise to not upgrade to latest version.

**Reply by krystian3w** [reply]
On Windows 7, chromium 110+ needs hacks to run.

**Reply by dportvine** [reply]
>Has anybody tested if the issue persists into 111-113?
<details>

```yaml
uBlock Origin: 1.47.4
Chromium: 111
filterset (summary): 
  network: 102872
  cosmetic: 129552
  scriptlet: 23801
  html: 0
listset (total-discarded, last-updated): 
  removed: 
    urlhaus-1: null
    plowe-0: null
  added: 
    https://easylist-downloads.adblockplus.org/bitblock.txt: 31568-26, 1h.10m
    https://easylist-downloads.adblockplus.org/cntblock.txt: 1551-26, 1h.10m
    https://www.awwwwesome.org/url-blocklist/url-blocklist.txt: 773-0, 1h.10m
    fanboy-cookiemonster: 45993-386, 1h.10m
    ublock-annoyances: 4845-52, 1h.10m
  default: 
    user-filters: 1-0, never
    ublock-filters: 32875-41, 1h.10m
    ublock-badware: 4640-0, 1h.10m
    ublock-privacy: 289-7, 1h.10m
    ublock-abuse: 78-0, 1h.10m
    ublock-unbreak: 1909-11, 1h.10m
    ublock-quick-fixes: 362-36, 1h.10m
    easylist: 63562-223, 1h.10m
    easyprivacy: 31023-1510, 1h.10m
    RUS-0: 39481-303, 1h.10m
filterset (user): [array of 2 redacted]
modifiedUserSettings: 
  advancedUserEnabled: true
  contextMenuEnabled: false
  showIconBadge: false
modifiedHiddenSettings: 
  cacheStorageAPI: indexedDB
  filterAuthorMode: true
  popupPanelHeightMode: 1
supportStats: 
  allReadyAfter: 3274 ms (selfie)
  maxAssetCacheWait: 536 ms
popupPanel: 
  blocked: 54
  blockedDetails: 
    github.com: 51
    google.com: 2
    translate.googleapis.com: 1
```
</details>

**Comment by gorhill** [comment]
I made changes in 1.47.5b2 which should help diagnose uBO's "unreadiness" at launch:

- At browser launch, uBO's toolbar icon will be yellow until it is ready to filter
- The number of unprocessed network requests at launch will be reported in troubleshooting info

**Reply by vlaszlo** [reply]
Would it be possible to start the extension in yellow, and only once change to red when it's ready? Because right now we see "red - yellow - red", which looks like "ready - not ready - ready", and that's probably not what's happening. It would also be slightly less distracting.

**Reply by krystian3w** [reply]
Probably only when there is an update from version to version in the background without disappearing from the bar for a millisecond…?

**Reply by vlaszlo** [reply]
The double color change you mean? No, I'm seeing it on every Firefox start, and the yellow phase takes about half a second.

**Reply by garry-ut99** [reply]
Confirmed, it's currently: dark red - yellow - dark red, in both Firefox & Opera.

<hr>

By the way

> https://github.com/uBlockOrigin/uBlock-issues/issues/2536#issuecomment-1465199304 + https://github.com/uBlockOrigin/uAssets/discussions/16939#discussioncomment-5346406 : red

Gwarser [would say](https://github.com/uBlockOrigin/uBlock-issues/issues/2519#issuecomment-1450467783) that he see dark red at most, not red, unlike certain users's semantic: calling a dark chocolate a chocolate, a grandmother a mother, an ad placeholder an ad, a space ship a ship, uBlock Origin - uBlock, Adblock Plus - Adblock.

**Reply by krystian3w** [reply]
Currently, ABP and AdBlock are driven by the same engine (~3 years), that if something doesn't work in one, it also doesn't works in the other if based on engine.

----

After an additional description, I now know how to look to notice color differences when they do not start with yellowish.

**Reply by gorhill** [reply]
> Would it be possible to start the extension in yellow

The browser picks the toolbar icon from the manifest before uBO is actually launched. Only after uBO is launched it can control the toolbar icon color. I did try to set the manifest default toolbar icons to yellowish unready ones, I still still the color changing back and forth at launch. For now we will have to live with this, I might look into this more in the next dev cycle.

**Reply by vlaszlo** [reply]
Thanks.

**Reply by uBlock-user** [reply]
>  still still the color changing back and forth at launch. For now we will have to live with this, I might look into this more in the next dev cycle.

All extensions icons are cached post installation. So to see new changes, that icon cache will need to be evicted too first.

**Reply by gorhill** [reply]
In that case if setting the not-ready icons by default, this could lead to an issue of the not-ready icon being used by the browser when merely opening a new tab, the same way the current default icon is used when opening a new webpage for which uBO is disabled or for priviledge pages. That would be worse than what we have now.

**Comment by ghost** [comment]
Wondering if this is related to filter lists load time or to something else.

Chrome (pretty clean):

```
[uBO] Admin settings ready 2830 ms after launch
[uBO] Backend storage for cache will be browser.storage.local
[uBO] Hidden settings ready 2838 ms after launch
[uBO] Extra admin settings ready 2839 ms after launch
[uBO] List selection ready 2839 ms after launch
[uBO] Cache magic numbers ready 2839 ms after launch
[uBO] First fetch ready 2842 ms after launch
[uBO] User settings ready 2844 ms after launch
[uBO] PSL ready 2852 ms after launch
[uBO] loadFilterLists() took 190 ms
[uBO] Filter lists ready 3130 ms after launch
[uBO] All ready 3131 ms after launch
```

Firefox (my dirty profile):

```
[uBO] Admin settings ready 26 ms after launch
[uBO] Backend storage for cache will be indexedDB
[uBO] Hidden settings ready 41 ms after launch
[uBO] Extra admin settings ready 44 ms after launch
[uBO] WASM modules ready 46 ms after launch
[uBO] List selection ready 60 ms after launch
[uBO] WASM PSL ready 61 ms after launch
[uBO] First fetch ready 64 ms after launch
[uBO] User settings ready 73 ms after launch
[uBO] Cache magic numbers ready 114 ms after launch
[uBO] PSL ready 124 ms after launch
[uBO] Selfie ready 343 ms after launch
[uBO] All ready 344 ms (selfie) after launch

```

**Reply by gorhill** [reply]
To make the case it's a `browser.storage`-related issue _at launch_, I [added timing information line](https://github.com/gorhill/uBlock/commit/d42e9ae2a890cb25c5244077ec6536926f3a149b) right before uBO starts to read from extension storage. This is the result:

    [uBO] Start sequence of loading storage-based data 41 ms after launch
    [uBO] Admin settings ready 2872 ms after launch
    [uBO] Backend storage for cache will be browser.storage.local
    [uBO] Hidden settings ready 2879 ms after launch
    [...]

**Reply by gorhill** [reply]
The regression is `browser.storage.managed`. If I bypass access to `browser.storage.managed`, I get:

    [uBO] Start sequence of loading storage-based data 29 ms after launch
    [uBO] Admin settings ready 31 ms after launch
    [uBO] Backend storage for cache will be browser.storage.local
    [uBO] Extra admin settings ready 37 ms after launch
    [uBO] List selection ready 42 ms after launch
    [uBO] Cache magic numbers ready 46 ms after launch
    [uBO] First fetch ready 47 ms after launch
    [uBO] User settings ready 48 ms after launch
    [uBO] PSL ready 55 ms after launch
    [uBO] Selfie ready 229 ms after launch
    [uBO] All ready 231 ms (selfie) after launch

**Reply by uBlock-user** [reply]
So to workaround this regression, `cacheStorageAPI` should be set to `indexedDB`  in Chrome ?

**Reply by gorhill** [reply]
That's not a solution, that will cause other problems we identified in the past when trying to use `indexedDB` in Chromium.

I will just add code to cache the content of `storage.managed` into `storage.local` and uBO will use this at launch time. `storage.managed` will still be called, but not in a blocking manner, and just to cache the content into `storage.local` for the next time uBO launches.

**Reply by uBlock-user** [reply]
Yes, that's why I said "workaround" this issue until a proper solution appears.

**Reply by ghost** [reply]
This fixes video ads on startup on YT for me 10 times out of 10 trials. And this without "Suspend network activity". Tried on `https://www.youtube.com/watch?v=LcmT38ZPTjg`. Good change. But now admins will complain about one more restart before policies will be active.

**Reply by stephenhawk8054** [reply]
@gorhill @gwarser Should we add a line in release announcement for warning the admins when updating to new version?

**Reply by gorhill** [reply]
We don't know if it will be different than before. For all I know, maybe the regression in Chromium 110 is because they fixed the `storage.managed` issue of having to restart, and me adding code to require this just bring back things as they were. We will have to wait for reports.

**Comment by uBlock-user** [comment]
1.47.5rc5 now throws an error in Nightly extension console -- 

![Capture](https://user-images.githubusercontent.com/21290713/226153101-3d2e96ee-8025-45df-adec-65385e3e0c8e.JPG)

**Reply by ghost** [reply]
I don't see this.

**Reply by uBlock-user** [reply]
I see it everytime I disable and then enable the extension. Last good build was 1.47.5rc2.

**Reply by ghost** [reply]
It's because I have actual manifest with `benchmarkDatasetURL` set.

**Reply by uBlock-user** [reply]
> I have actual manifest

???

**Reply by gorhill** [reply]
See if this is fixed in 1.47.5rc6.

**Reply by uBlock-user** [reply]
Yep, back to normal with rc6.

**Comment by jline9507** [comment]
The Google Chrome browser is the better web browser for general internet use. If you're a developer[,](https://bettafishinfo.com/best-plants-for-betta-fish/) Chromium is a better resource for testing new features or creating a new browser. Is Chromium Safer Than Chrome? The standard version of Chromium is less secure than Chrome and equally private

**Reply by krystian3w** [reply]
But nobody asked.

**Comment by krystian3w** [comment]
How this is spam: 
![obraz](https://github.com/uBlockOrigin/uAssets/assets/35370833/40a29fab-3e9c-47bc-a897-bd7e447d5eed)
 
GPT bot spamed, but not me.
